moveit_core/planning_interface/include/moveit/planning_interface/planning_response.h

```diff
@@ -37,23 +37,38 @@
 #pragma once
 
 #include <moveit/robot_trajectory/robot_trajectory.h>
+#include <moveit/utils/moveit_error_code.h>
 #include <moveit_msgs/msg/move_it_error_codes.hpp>
 #include <moveit_msgs/msg/motion_plan_response.hpp>
 #include <moveit_msgs/msg/motion_plan_detailed_response.hpp>
 
 namespace planning_interface
 {
+/// \brief Response to a planning query
 struct MotionPlanResponse
 {
-  MotionPlanResponse() : planning_time_(0.0)
+  // Constructor
+  MotionPlanResponse() : trajectory_(nullptr), planning_time_(0.0), error_code_(moveit::core::MoveItErrorCode::FAILURE)
   {
   }
 
+  // Construct a ROS message from struct data
   void getMessage(moveit_msgs::msg::MotionPlanResponse& msg) const;
 
+  // Trajectory generated by the queried planner
   robot_trajectory::RobotTrajectoryPtr trajectory_;
+  // Time to plan the respond to the planning query
   double planning_time_;
-  moveit_msgs::msg::MoveItErrorCodes error_code_;
+  // Result status of the query
+  moveit::core::MoveItErrorCode error_code_;
```

Comment:
This change is not consistent with `MotionPlanDetailedResponse`. Whatever you change here should be reflected in the other struct as well.

Reply:
I'll add it there as well :+1:

Reply:
@henningkayser What do you think about using inheritance
```cpp
struct MotionPlanDetailedResponse : public MotionPlanResponse
{
  // Make this a free function for both classes
  // void getMessage(moveit_msgs::msg::MotionPlanDetailedResponse& msg) const; 

  std::vector<std::string> description_;
  std::vector<double> processing_time_;
};
```
or aggregation here?
```cpp
struct MotionPlanDetailedResponse : public MotionPlanResponse
{
   void getMessage(moveit_msgs::msg::MotionPlanDetailedResponse& msg) const; 

  MotionPlanResponse response;
  std::vector<std::string> description_;
  std::vector<double> processing_time_;
};
```
I think both would avoid changing the other if one changes in the future. Or to remove `MotionPlanDetailedResponse` completely and make processing time and description `std::optional`

Reply:
well, this got stale... I think neither option is applicable, the detailed response includes a vector of trajectories.

Reply:
I see, I'll update this PR

Reply:
The message definition states
```
# Multiple solution paths are reported, each reflecting intermediate steps in the trajectory processing

# The list of reported trajectories
```
But is that used anywhere/ how is it interpretable when the response is just a vector of unlabeled trajectories?

Reply:
I'm not sure if it's being used anywhere, but my understanding is that the trajectories show you the result for each step in the process. For instance, you should be able to compare the raw planner solution with the smoothed output.